src/main/java/revai/StreamingClient.java

```diff
@@ -0,0 +1,300 @@
+package revai;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.WebSocket;
+import okhttp3.WebSocketListener;
+import okio.ByteString;
+import revai.helpers.SDKHelper;
+import revai.models.streaming.Hypothesis;
+
+public class StreamingClient {
+
+  private String accessToken;
+  private OkHttpClient client;
+  private WebSocket webSocket;
+  private String scheme;
+  private String host;
+  private Integer port;
+
+  /**
+   * Constructs the streaming client that is used to establish a WebSocket connection with the
+   * Rev.ai server and stream audio data. The user access token can be generated on the website at
+   * https://www.rev.ai/access_token.
+   *
+   * @param accessToken Rev.ai authorization token.
+   * @see StreamContentType
+   */
+  public StreamingClient(String accessToken) {
+    this.accessToken = accessToken;
+    this.client = setClient();
+  }
+
+  /**
+   * This method sets the URL scheme to be used in the WebSocket connect request
+   *
+   * @param scheme URL scheme.
+   */
+  public void setScheme(String scheme) {
+    this.scheme = scheme;
+  }
+
+  /**
+   * This methods sets the URL host name. By default the host name is api.rev.ai.
+   *
+   * @param host URL host name.
+   */
+  public void setHost(String host) {
+    this.host = host;
+  }
+
+  /**
+   * This method sets a port number to be used in the WebSocket connect request.
+   *
+   * @param port the port used to connect to
+   */
+  public void setPort(int port) {
+    this.port = port;
+  }
+
+  /**
+   * Sends an HTTP request and upon authorization is upgraded to a WebSocket connection. Use
+   * {@link RevAiWebSocketListener} to handle web socket events. To establish a successful connection a
+   * valid StreamContentType must be provided.
+   *
+   * <p>Providing metadata is optional but helps to identify the stream.
+   *
+   * <p>Providing a custom vocabulary is optional. This method assigns a custom vocabulary to be
+   * used during the stream. Custom vocabularies are submitted prior to usage in the stream and
+   * assigned an Id.
+   *
+   * <p>Setting the profanity filter is optional. More information regarding the profanity filter
+   * can be found here: <a
+   * href="https://www.rev.ai/docs/streaming#section/WebSocket-Endpoint/Filter-Profanity">Filter
+   * Profanity</a
+   *
+   * @param revAiWebSocketListener the listener used to capture WebSocket events.
+   * @param metadata used to identify the stream.
+   * @param streamContentType content-type query parameter.
+   * @param customVocabularyId customVocabularyId the ID of the custom vocabulary.
+   * @param filterProfanity this is false by default.
+   * @see RevAiWebSocketListener
+   * @see StreamContentType
+   */
+  public void connect(
+      RevAiWebSocketListener revAiWebSocketListener,
+      String metadata,
```

Comment:
I would put metadata, custom vocab id, and filter profanity into a `SessionConfig` class. then we don't need so many overloads, just two: one that takes only the content type, and one that takes content type and sessionConfig. all fields of this sessionConfig would be optional.

so eg one way to call would be 

```
connect(listener,
    new StreamContentType. { ContentType = "audio/mp3" }, 
    new SessionConfig { filterProfanity = true }
);
```